src/query/storage/query_conversion_lru.go

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) 2019 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package storage
+
+import (
+	"container/list"
+	"fmt"
+
+	"github.com/m3db/m3/src/m3ninx/idx"
+)
+
+// QueryConversionLRU implements a fixed size LRU cache
+type QueryConversionLRU struct {
+	size      int
+	evictList *list.List
+	items     map[string]*list.Element
+}
+
+// entry is used to hold a value in the evictList
+type entry struct {
+	key   string
+	value idx.Query
+}
+
+// NewQueryConversionLRU constructs an LRU of the given size
+func NewQueryConversionLRU(size int) (*QueryConversionLRU, error) {
+	if size <= 0 {
+		return nil, fmt.Errorf("must provide a positive size, instead got: %d", size)
+	}
+
+	c := &QueryConversionLRU{
+		size:      size,
+		evictList: list.New(),
+		items:     make(map[string]*list.Element, size),
+	}
+
+	return c, nil
+}
+
+// Add adds a value to the cache. Returns true if an eviction occurred.
+func (c *QueryConversionLRU) Add(key string, value idx.Query) (evicted bool) {
```

Comment:
Can you keep key as a []byte here, then cast it only when indexing into the map?